src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseStringEqualsOverStringCompare.Fixer.cs

```diff
@@ -0,0 +1,322 @@
+            /// </summary>
+            /// <param name="compareResultToLiteralZero">An operation that is potentially a comparison of an invocation with a literal zero.</param>
+            /// <param name="invocation">The invocation operation.</param>
+            /// <returns>True if the specified operation is an equals or not-equals operation that compares a literal zero to 
+            /// any invocation operation.</returns>
+            protected static bool TryGetInvocationFromComparisonWithLiteralZero(IOperation compareResultToLiteralZero, [NotNullWhen(true)] out IInvocationOperation? invocation)
+            {
+                if (compareResultToLiteralZero is IBinaryOperation binaryOperation &&
+                    binaryOperation.OperatorKind is BinaryOperatorKind.Equals or BinaryOperatorKind.NotEquals)
+                {
+                    if (TryConvertOperands(binaryOperation.LeftOperand, binaryOperation.RightOperand, out invocation) ||
+                        TryConvertOperands(binaryOperation.RightOperand, binaryOperation.LeftOperand, out invocation))
+                    {
+                        return true;
+                    }
+                }
+
+                invocation = default;
+                return false;
+
+                //  Local functions
+
+                static bool TryConvertOperands(IOperation first, IOperation second, [NotNullWhen(true)] out IInvocationOperation? result)
+                {
+                    if (first is IInvocationOperation invocation &&
+                        second is ILiteralOperation literal &&
+                        literal.ConstantValue.HasValue &&
+                        literal.ConstantValue.Value is int integer &&
+                        integer is 0)
+                    {
+                        result = invocation;
+                        return true;
+                    }
+
+                    result = default;
+                    return false;
+                }
+            }
+
+            protected static IInvocationOperation GetInvocationFromComparisonWithLiteralZero(IOperation compareResultToLiteralZero)
+            {
+                if (!TryGetInvocationFromComparisonWithLiteralZero(compareResultToLiteralZero, out var invocation))
+                    Fail();
+
+                return invocation;
+            }
+
+            protected static bool IsNotEqualsOperation(IOperation operation)
+            {
+                return operation is IBinaryOperation binaryOperation &&
+                    binaryOperation.OperatorKind is BinaryOperatorKind.NotEquals;
+            }
+
+            [DoesNotReturn]
+#pragma warning disable CS8763 // A method marked [DoesNotReturn] should not return.
+            protected static void Fail() => Debug.Fail($"'{nameof(GetReplacementExpression)}' must only be called when '{nameof(IsMatch)}' is 'true'.");
+#pragma warning restore CS8763 // A method marked [DoesNotReturn] should not return.
```

Comment:
If the failure is intended for debug only, you can add `[Conditional("DEBUG")]` attribute.